float64 pos_x
float64 pos_y
float64 pos_z
int32 error_type
bool stop